js/src/elements/star-rating.ts

```diff
@@ -238,18 +252,27 @@ export class StarRatingElement extends HTMLElement {
 		return parseInt(this.getAttribute('value') ?? '');
 	}
 
-	/** TODO: add form related properties (i.e validity, etc */
+	get validity() {
+		// borrow radio input's native validity
+		return this.shadowRoot?.querySelector('input')?.validity;
+	}
 
 	connectedCallback() {
 		this.shadowRoot?.addEventListener('change', this);
```

Comment:
future pr but it looks like when clicking on a the "required" validation message in the form behavior's validation summary the star rating is focused but the focus doesn't quite get to the input itself. We may need to add an event listener to the star-rating for `focus` and in turn call focus on the checked input, falling back to the first input.

Reply:
Good catch. I explored some options yesterday, will either add to this PR or another PR depending on timing.